shared/protocol/src/test/java/io/pravega/shared/protocol/netty/WireCommandsTest.java

```diff
@@ -894,6 +894,16 @@ public void testConditionalBlockEnd() throws IOException {
                 ce -> ce.getData().refCnt());
     }
 
+    @Test
+    public void testMergeSegmentsWithAttributes() throws IOException {
+        List<WireCommands.ConditionalAttributeUpdate> attributeUpdates = Arrays.asList(
```

Comment:
Please add a unit test to verify backwards compatibility (like when you receive a serialization that does not include attributes).

Reply:
Added a test called `testCompatibilityMergeSegmentsV5()` to cover that case.